third_party/move/move-vm/types/src/values/serialization_tests.rs

```diff
@@ -53,3 +63,70 @@ fn enum_round_trip() {
         })
         .expect_err("bad struct value deserialization fails");
 }
+
+#[test]
+fn test_serialized_size() {
```

Comment:
If time permits, it might be helpful to have some proptests here to ensure the invariant `len(serialize(val)) = serialized_size(val)` always holds

Reply:
Adding this to https://github.com/aptos-labs/aptos-core/issues/14175 ticket, will do in the meantime, but this should not block this from landing